src/ioctls/vcpu.rs

```diff
@@ -1220,18 +1226,20 @@ impl VcpuFd {
     ///
     /// * `reg_id` - ID of the register.
     #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
-    pub fn get_one_reg(&self, reg_id: u64) -> Result<u64> {
-        let mut reg_value = 0;
+    pub fn get_one_reg<T: ValidRegisterType>(&self, reg_id: u64) -> Result<T> {
+        let mut reg_value = std::mem::MaybeUninit::<T>::uninit();
```

Comment:
I think we still need to check the size of `T` here to make sure the function does not cause memory safety issues.

Reply:
That's a good idea. Alternatively we can initialize this value with 0s because if the user requests a register with size larger that the size that the kernel actually writes, we will end up with garbage data in the result.